### PR TITLE
Feat/update JupyterLab

### PR DESCRIPTION
**Pull request by ricklamers**
## Description

This PR will update JupyterLab to 3.2.4. As a prerequisite we're also updating Jupyter Enterprise Gateway to 2.5.x to fix an issue with the kernelspec resource handling.

Issue that is blocking this: https://github.com/jupyterlab/jupyterlab/issues/11527

### Checklist

- [x] The documentation reflects the changes.
- [x] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

### CODE REVIEW

File: services/orchest-api/app/app/core/sessions.py [hunk 1]
```diff
@@ -1112,6 +1114,11 @@ def _get_orchest_services_specs(
             jupyer_server_image = user_jupyer_server_image
 
         # Run Jupyter server container.
+        base_url = "/%s" % _config.JUPYTER_SERVER_NAME.format(
+            project_uuid=project_uuid[: _config.TRUNCATED_UUID_LENGTH],
+            pipeline_uuid=pipeline_uuid[: _config.TRUNCATED_UUID_LENGTH],
+        )
+
```
Comment (fruttasecca) [inline services/orchest-api/app/app/core/sessions.py:1121]: I see that the same assignment exists at line 1064, there might be something I am not seeing about it.

File: services/orchest-api/app/app/core/sessions.py [hunk 2]
```diff
@@ -1075,6 +1081,7 @@ def _get_orchest_services_specs(
                 'EG_UNAUTHORIZED_USERS=["dummy"]',
                 'EG_UID_BLACKLIST=["-1"]',
                 "EG_ALLOW_ORIGIN=*",
+                "EG_BASE_URL=%s" % base_url,
```
Comment (fruttasecca) [inline services/orchest-api/app/app/core/sessions.py:1084]: Is this needed by the new version? Just trying to understand the changes :).

**Review by fruttasecca — APPROVED** [review]
lgtm

File: services/orchest-api/app/app/core/sessions.py (hunk 2 above)
Comment (ricklamers) [reply services/orchest-api/app/app/core/sessions.py:1085]: Yeah, this assignment is unnecessarily duplicated. Will commit a deletion

**Pull request merged**